content/intro-to-storybook/react-native/en/simple-component.md

```diff
@@ -0,0 +1,269 @@
+        />
+      </TouchableOpacity>
+    </SafeAreaView>
+  );
+}
+```
+
+The additional markup from above combined with the styling we created earlier yields the following UI:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook/finished-task-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## Specify data requirements
+
+It’s best practice to use `propTypes` in React to specify the shape of data that a component expects. Not only is it self documenting, it also helps catch problems early.
+
+```javascript
+// src/components/Task.js
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Task({ task: { id, title, state }, onArchiveTask, onPinTask }) {
+  // ...
+}
+
+Task.propTypes = {
+  task: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    title: PropTypes.string.isRequired,
+    state: PropTypes.string.isRequired,
+  }),
+  onArchiveTask: PropTypes.func,
+  onPinTask: PropTypes.func,
+};
+```
+
+Now a warning in development will appear if the Task component is misused.
+
+<div class="aside">
+An alternative way to achieve the same purpose is to use a JavaScript type system like TypeScript to create a type for the component properties.
+</div>
+
+## Component built!
+
+We’ve now successfully built out a component without needing a server or running the entire frontend application. The next step is to build out the remaining Taskbox components one by one in a similar fashion.
+
+As you can see, getting started building components in isolation is easy and fast. We can expect to produce a higher-quality UI with fewer bugs and more polish because it’s possible to dig in and test every possible state.
+
+## Automated Testing
+
+Storybook gave us a great way to visually test our application during construction. The ‘stories’ will help ensure we don’t break our Task visually as we continue to develop the app. However, it is a completely manual process at this stage, and someone has to go to the effort of clicking through each test state and ensuring it renders well and without errors or warnings. Can’t we do that automatically?
+
+### Snapshot testing
+
+Snapshot testing refers to the practice of recording the “known good” output of a component for a given input and then flagging the component whenever the output changes in future. This complements Storybook, because it’s a quick way to view the new version of a component and check out the changes.
+
+<div class="aside">
+Make sure your components render data that doesn't change, so that your snapshot tests won't fail each time. Watch out for things like dates or randomly generated values.
+</div>
+
+With the [Storyshots addon](https://github.com/storybooks/storybook/tree/master/addons/storyshots) a snapshot test is created for each of the stories. Use it by adding a development dependency on the package:
+
+```bash
+yarn add --dev @storybook/addon-storyshots babel-plugin-require-context-hook
```

Comment:
I don't think you are using `require.context()` above so this step shouldn't be necessary?

Reply:
I was using it for some testing purposes with the create addon section. As to test if a error i was getting was actually related to that package and left it in. Removed it in the latest commit